app/src/backend/ActivityService.ts

```diff
@@ -217,7 +219,7 @@ export default class ActivityService {
           BigInt(inventoryValue.co2eq as bigint) -
             BigInt(activityValue.co2eq as bigint) ?? 0n;
 
-        const calculatedCO2e = BigInt(totalCO2e); // Ensure totalCO2e is BigInt
+        const calculatedCO2e = BigInt(totalCO2e.toNumber()); // Ensure totalCO2e is BigInt
```

Comment:
Can you instead convert totalCO2e to a string and pass it here? Otherwise you're losing all of the precision gained by using decimal.js here by using the lossy JS Number type as an intermediary.